modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/portlet/FDSSamplePortlet.java

```diff
@@ -81,22 +82,22 @@ public void doDispatch(
 			RenderRequest renderRequest, RenderResponse renderResponse)
 		throws IOException, PortletException {
 
-		renderRequest.setAttribute(
-			FDSSampleWebKeys.FDS_SAMPLE_DISPLAY_CONTEXT,
-			new FDSSampleDisplayContext(
-				_portal.getHttpServletRequest(renderRequest)));
-
-		super.doDispatch(renderRequest, renderResponse);
-	}
+		ThemeDisplay themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
 
-	@Activate
-	protected void activate() {
 		try {
-			_generate(_portal.getDefaultCompanyId());
+			_generate(themeDisplay.getCompanyId());
```

Comment:
back to the origins :) 
LGTM this looks much more safer for the stability of the testing even if that forces some delays in the first render